### PR TITLE
채팅방 생성 기능

### LINKED ISSUE #109: 채팅방 생성
## AS-IS
현재 채팅방을 생성하는 API가 개발되어 있지 않다.

## TO-BE
채팅방을 생성하는 기능을 개발한다.

### PR DESCRIPTION
**Pull request by 23Yong**
## Issues
- #109 

## What is this PR? 👓
채팅방 생성 기능에 대한 PR입니다.

## Key changes 🔑
- 채팅방 생성
- 아이템의 채팅방 카운트 증가
- 테스트코드 작성

## To reviewers 👋
아이템의 채팅방 카운트는 별도의 이벤트로 뺄 이유가 없어 같은 로직에서 처리했습니다.
왜냐하면 같은 쓰기 트랜잭션이고 로직에서 `item`을 가지고 있어 같은 로직에서 처리했습니다.

### CODE REVIEW

**Comment by 23Yong** [comment]
### 📝 테스트 커버리지 리포트
|File|Coverage [76.05%]|:green_apple:|
|:-|:-:|:-:|
|[ChatRoomService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/7124d89556e948d008c4c4ed62955cf633c6fb16/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fchat%2FChatRoomService.java)|93.75%|:green_apple:|
|[Item.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/7124d89556e948d008c4c4ed62955cf633c6fb16/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fitem%2FItem.java)|91.64%|:green_apple:|
|[ChatController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/7124d89556e948d008c4c4ed62955cf633c6fb16/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FChatController.java)|36.5%|:x:|

|Total Project Coverage|79.92%|:green_apple:|
|:-|:-:|:-:|

**Comment by 23Yong** [comment]
### 📝 테스트 커버리지 리포트
|File|Coverage [76.36%]|:green_apple:|
|:-|:-:|:-:|
|[ChatRoomService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/be167c4996fa097e863e448bdd904e32b40a3d53/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fchat%2FChatRoomService.java)|93.75%|:green_apple:|
|[Item.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/be167c4996fa097e863e448bdd904e32b40a3d53/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fitem%2FItem.java)|91.64%|:green_apple:|
|[ChatRoom.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/be167c4996fa097e863e448bdd904e32b40a3d53/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fchat%2FChatRoom.java)|77.52%|:green_apple:|
|[ChatController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/be167c4996fa097e863e448bdd904e32b40a3d53/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FChatController.java)|36.5%|:x:|

|Total Project Coverage|79.87%|:green_apple:|
|:-|:-:|:-:|

**Comment by 23Yong** [comment]
### 📝 테스트 커버리지 리포트
|File|Coverage [76.36%]|:green_apple:|
|:-|:-:|:-:|
|[ChatRoomService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/a85d05ed7da7a3087419ce6a10075701c77f33e7/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fchat%2FChatRoomService.java)|93.75%|:green_apple:|
|[Item.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/a85d05ed7da7a3087419ce6a10075701c77f33e7/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fitem%2FItem.java)|91.64%|:green_apple:|
|[ChatRoom.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/a85d05ed7da7a3087419ce6a10075701c77f33e7/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fchat%2FChatRoom.java)|77.52%|:green_apple:|
|[ChatController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/a85d05ed7da7a3087419ce6a10075701c77f33e7/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FChatController.java)|36.5%|:x:|

|Total Project Coverage|79.87%|:green_apple:|
|:-|:-:|:-:|

**Comment by 23Yong** [comment]
### 📝 테스트 커버리지 리포트
|File|Coverage [76.24%]|:green_apple:|
|:-|:-:|:-:|
|[ChatRoomService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/2ad0bcf141bf6ab6fda68bd5675c7a9c4204e602/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fchat%2FChatRoomService.java)|93.75%|:green_apple:|
|[Item.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/2ad0bcf141bf6ab6fda68bd5675c7a9c4204e602/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fitem%2FItem.java)|91.64%|:green_apple:|
|[ChatRoom.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/2ad0bcf141bf6ab6fda68bd5675c7a9c4204e602/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fchat%2FChatRoom.java)|77.52%|:green_apple:|
|[ChatController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/2ad0bcf141bf6ab6fda68bd5675c7a9c4204e602/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FChatController.java)|36.23%|:x:|

|Total Project Coverage|79.86%|:green_apple:|
|:-|:-:|:-:|

**Comment by 23Yong** [comment]
### 📝 테스트 커버리지 리포트
|File|Coverage [77.68%]|:green_apple:|
|:-|:-:|:-:|
|[MemberController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/9ddd96ae7a568c9d1bd41a85a100331515d2364d/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FMemberController.java)|100%|:green_apple:|
|[ChatRoomService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/9ddd96ae7a568c9d1bd41a85a100331515d2364d/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fchat%2FChatRoomService.java)|93.75%|:green_apple:|
|[Item.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/9ddd96ae7a568c9d1bd41a85a100331515d2364d/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fitem%2FItem.java)|91.64%|:green_apple:|
|[MemberService.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/9ddd96ae7a568c9d1bd41a85a100331515d2364d/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fapplication%2Fmember%2FMemberService.java)|86.67%|:green_apple:|
|[ChatRoom.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/9ddd96ae7a568c9d1bd41a85a100331515d2364d/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fdomain%2Fchat%2FChatRoom.java)|77.52%|:green_apple:|
|[ChatController.java](https://github.com/masters2023-project-03-second-hand/second-hand-max-be-b/blob/9ddd96ae7a568c9d1bd41a85a100331515d2364d/src%2Fmain%2Fjava%2Fkr%2Fcodesquad%2Fsecondhand%2Fpresentation%2FChatController.java)|36.23%|:x:|

|Total Project Coverage|79.88%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/kr/codesquad/secondhand/application/member/MemberService.java [hunk 1]
```diff
@@ -23,16 +23,17 @@ public class MemberService {
     private String defaultProfileImage;
 
     @Transactional
-    public void modifyProfileImage(MultipartFile profileImage, Long memberId) {
+    public String modifyProfileImage(MultipartFile profileImage, Long memberId) {
```
Comment (he2joojo) [inline src/main/java/kr/codesquad/secondhand/application/member/MemberService.java:26]: 바로 반영해주셨네요!👍

**Review by he2joojo — APPROVED** [review]
수고하셨습니다~!

**Pull request merged**